### ISSUE TITLE
Download stops after first blob, payment sent, restarts in stopped status

### ISSUE THREAD

**Issue by tzarebczan** [lbryio/lbry#1172]
## The Issue
This has happened to me twice (update, now 3 times!) now, not sure how to reproduce it consistently.  

```
2018-04-27 15:08:20,618 INFO     lbrynet.daemon.Downloader:93: Download lbry://two status changed to initializing
2018-04-27 15:08:20,624 INFO     lbrynet.daemon.ExchangeRateManager:221: Converting 50.000000 LBC to LBC, rates: [Currency pair:('USD', 'BTC'), spot:0.000108066589509, ts:1524856044, Currency pair:('BTC', 'LBC'), spot:29350.7670639, ts:1524856044, Currency pair:('BTC', 'LBC'), spot:29351.9505936, ts:1524856044, Currency pair:('USD', 'BTC'), spot:0.00010799, ts:1524855145, Currency pair:('BTC', 'LBC'), spot:29239.7660819, ts:1524856045]
2018-04-27 15:08:20,624 INFO     lbrynet.daemon.ExchangeRateManager:221: Converting 10.000000 LBC to LBC, rates: [Currency pair:('USD', 'BTC'), spot:0.000108066589509, ts:1524856044, Currency pair:('BTC', 'LBC'), spot:29350.7670639, ts:1524856044, Currency pair:('BTC', 'LBC'), spot:29351.9505936, ts:1524856044, Currency pair:('USD', 'BTC'), spot:0.00010799, ts:1524855145, Currency pair:('BTC', 'LBC'), spot:29239.7660819, ts:1524856045]
2018-04-27 15:08:20,625 INFO     lbrynet.daemon.Downloader:93: Download lbry://two status changed to downloading_metadata
2018-04-27 15:08:20,973 INFO     lbrynet.core.client.BlobRequester:414: Offered rate 0.000000/mb accepted by 18.221.194.193
2018-04-27 15:08:20,973 INFO     lbrynet.core.client.BlobRequester:487: Requesting blob f204bb18354b803c17ecbfd86e796a3ec4fd0097506812a97189a6ffe6f2a9f41dcfd02463f33a587015ffef870bf0fe from 18.221.194.193:3333
2018-04-27 15:08:21,086 INFO     lbrynet.core.client.BlobRequester:563: Blob f204bb18354b803c has been successfully downloaded from 18.221.194.193:3333
2018-04-27 15:08:21,638 INFO     lbrynet.daemon.Downloader:132: Pay key fee 10.000000 --> bVxoPVM3HmbiE2yhB8RsFcViSK6RSj6kX9
2018-04-27 15:08:21,654 INFO     lbrynet.daemon.Downloader:186: Downloading lbry://two (f204bb) --> C:\Users\SoloDolo\Downloads\lbry downloads\Mystery-Box-1.zip
2018-04-27 15:08:21,654 INFO     lbrynet.daemon.Downloader:93: Download lbry://two status changed to running
2018-04-27 15:08:21,934 INFO     lbrynet.core.client.BlobRequester:414: Offered rate 0.000000/mb accepted by 18.221.194.193
2018-04-27 15:08:21,934 INFO     lbrynet.core.client.BlobRequester:487: Requesting blob 8341e24dc7b56493d0d941830466fa93b0ed5f449c4cf3a50e8c6db4885650de7e7d216fc11d87ec80cd23cd6f7ffc45 from 18.221.194.193:3333
2018-04-27 15:09:00,469 INFO     lbrynet.core.Wallet:310: Skipping dust
2018-04-27 15:09:00,549 INFO     root:205: using buckets: [u'ff4ed37a622dfbd9eb873613d218b2d1a2c3a8053f0288bdd66e5c26f9ef19db:0', u'7c75918f31a2df058ae633b27c9625c31bb4dc106725d414d7c25dd1dcf2e075:1']
```
Nothing happens after the last blob is requested. When restarting daemon, the file is in stopped status so it doensn't  ever complete downloading. Running a file_set_status start on the sd_hash it finishes. 

```
  "blobs_completed": null,
    "blobs_in_stream": null,
    "channel_claim_id": "deeb66c6f87e637747f88028ce77d16b9ba67d5e",
    "channel_name": "@mystery-box",
    "claim_id": "eee58ceec76059fd2d6b36dbddf0b5faca408130",
    "claim_name": "two",
    "completed": false,
    "download_directory": "C:\\Users\\xxx\\Downloads\\lbry downloads",
    "download_path": "C:\\Users\\xxx\\Downloads\\lbry downloads\\Mystery-Box-1.zip",
    "file_name": "Mystery-Box-1.zip",
    "key": "a075f4138365a717e3f924e07e428b25",
    "metadata": {
      "author": "",
      "description": "Secret Mystery Box - What goodies are inside? Find out.\n\n\nSometimes these boxes contain amazing gifts, sometimes normal gifts and sometimes nothing.",
      "fee": {
        "address": "bVxoPVM3HmbiE2yhB8RsFcViSK6RSj6kX9",
        "amount": 10.0,
        "currency": "LBC",
        "version": "_0_0_1"
      },
      "language": "en",
      "license": "Public Domain",
      "licenseUrl": "",
      "nsfw": false,
      "preview": "",
      "thumbnail": "https://spee.ch/6/mystery-box.jpeg",
      "title": "Secret Mystery Box",
      "version": "_0_1_0"
    },
    "mime_type": "application/x-zip-compressed",
    "nout": 0,
    "outpoint": "040e43e33d55bd7aba2ec0e1c355206de18ce031e2ca964a7c5068b3a1f56a9e:0",
    "points_paid": 0.0,
    "sd_hash": "f204bb18354b803c17ecbfd86e796a3ec4fd0097506812a97189a6ffe6f2a9f41dcfd02463f33a587015ffef870bf0fe",
    "status": null,
    "stopped": true,
    "stream_hash": "08657cbdfbd74a15f4ab68b915a42a808aa059a35116783d3e7a5088750e65f2c6ac3df1f8bda0725e2d876b2832746f",
    "stream_name": "Mystery-Box-1.zip",
    "suggested_file_name": "Mystery-Box-1.zip",
    "total_bytes": null,
    "txid": "040e43e33d55bd7aba2ec0e1c355206de18ce031e2ca964a7c5068b3a1f56a9e",
    "written_bytes": 0
  }
```

### Steps to reproduce
1.
2.
3.

### Expected behaviour
Tell us what should happen

### Actual behaviour
Tell us what happens instead

## System Configuration

 

- LBRY Daemon version: 0.19.1
- LBRY App version:
- LBRY Installation ID:
- Operating system:

**Comment by tzarebczan** [comment]
Happened to me again:
```
2018-05-31 17:30:13,914 INFO     lbrynet.daemon.Downloader:93: Download lbry://theperfecthouse status changed to downloading_metadata
2018-05-31 17:30:14,579 INFO     lbrynet.core.client.BlobRequester:414: Offered rate 0.000000/mb accepted by 47.17.162.103
2018-05-31 17:30:14,581 INFO     lbrynet.core.client.BlobRequester:487: Requesting blob 409a214fbc412ccbe87c584340bbb150eb0cc9a8e81d090ff47ed0c469c4b8ba21be25b3593d35d5938a54a1d711c03a from 47.17.162.103:3352
2018-05-31 17:30:14,588 INFO     lbrynet.core.client.BlobRequester:414: Offered rate 0.000000/mb accepted by 18.191.37.145
2018-05-31 17:30:14,588 INFO     lbrynet.core.client.BlobRequester:487: Requesting blob 409a214fbc412ccbe87c584340bbb150eb0cc9a8e81d090ff47ed0c469c4b8ba21be25b3593d35d5938a54a1d711c03a from 18.191.37.145:3333
2018-05-31 17:30:14,609 INFO     lbrynet.core.client.BlobRequester:563: Blob 409a214fbc412ccb has been successfully downloaded from 47.17.162.103:3352
2018-05-31 17:30:14,634 INFO     lbrynet.core.client.ClientProtocol:185: Closing the connection to 18.191.37.145:3333 because the download of blob 409a214fbc412ccb was canceled
2018-05-31 17:30:15,566 INFO     lbrynet.daemon.Daemon:1678: Already waiting on lbry://theperfecthouse to start downloading
2018-05-31 17:30:16,563 INFO     lbrynet.daemon.Downloader:132: Pay key fee 5.405147 --> bCvxVJX92FQ54UVN2BbbUeYUqVJusf5ynL
2018-05-31 17:30:16,573 INFO     lbrynet.daemon.Downloader:186: Downloading lbry://theperfecthouse (409a21) --> c:\lbry4\Downloads\The Perfect House 1080.mp4
2018-05-31 17:30:16,575 INFO     lbrynet.daemon.Downloader:93: Download lbry://theperfecthouse status changed to running
2018-05-31 17:30:16,875 INFO     lbrynet.core.client.BlobRequester:414: Offered rate 0.000000/mb accepted by 47.17.162.103
2018-05-31 17:30:16,881 INFO     lbrynet.core.client.BlobRequester:487: Requesting blob 3db2f9b5626f705fb22b0034ea99bd548871a25086e69cd9ef5bfee6ca4d787422f53467cc70cecc4f10d668d653c8ef from 47.17.162.103:3352
2018-05-31 17:31:06,674 INFO     lbrynet.core.Wallet:395: Skipping dust
2018-05-31 17:31:06,674 INFO     lbrynet.core.Wallet:395: Skipping dust
```

No timeout, sitting in stopped = true, completed =false

Upon shutting down the daemon, did get this:
```
Unhandled error in Deferred:
2018-05-31 18:03:25,729 CRITICAL twisted:154: Unhandled error in Deferred:

2018-05-31 18:03:25,747 CRITICAL twisted:154:
Traceback (most recent call last):
  File "site-packages\twisted\internet\defer.py", line 1299, in _inlineCallbacks
  File "site-packages\lbrynet\daemon\Daemon.py", line 671, in _download_failed
  File "site-packages\lbrynet\analytics.py", line 61, in send_download_errored
  File "site-packages\lbrynet\analytics.py", line 182, in _download_error_properties
AttributeError: Failure instance has no attribute 'message'
```

**Comment by tzarebczan** [comment]
Happened again:

```
2018-06-17 13:29:16,984 INFO     lbrynet.daemon.Downloader:93: Download lbry://three status changed to initializing
2018-06-17 13:29:16,987 INFO     lbrynet.daemon.ExchangeRateManager:228: Converting 50.000000 LBC to LBC, rates: [Currency pair:('USD', 'BTC'), spot:0.000153817283525, ts:1529256465, Currency pair:('BTC', 'LBC'), spot:56575.2701717, ts:1529256465, Currency pair:('BTC', 'LBC'), spot:56582.3860262, ts:1529256466]
2018-06-17 13:29:16,987 INFO     lbrynet.daemon.ExchangeRateManager:228: Converting 10.000000 LBC to LBC, rates: [Currency pair:('USD', 'BTC'), spot:0.000153817283525, ts:1529256465, Currency pair:('BTC', 'LBC'), spot:56575.2701717, ts:1529256465, Currency pair:('BTC', 'LBC'), spot:56582.3860262, ts:1529256466]
2018-06-17 13:29:16,987 INFO     lbrynet.daemon.Downloader:93: Download lbry://three status changed to downloading_metadata
2018-06-17 13:30:50,392 INFO     lbrynet.core.client.BlobRequester:414: Offered rate 0.000000/mb accepted by 18.188.64.47
2018-06-17 13:30:50,392 INFO     lbrynet.core.client.BlobRequester:487: Requesting blob 348d5fd8faa643bfd96e5eb69196c3e2969e48f12bbac1aa4f1f6536a6dea4f1ed943f51ae758f7ee54a7371e136c659 from 18.188.64.47:3333
2018-06-17 13:30:50,430 INFO     lbrynet.core.client.BlobRequester:563: Blob 348d5fd8faa643bf has been successfully downloaded from 18.188.64.47:3333
2018-06-17 13:30:51,081 INFO     lbrynet.daemon.Downloader:132: Pay key fee 10.000000 --> bZYUVLcKXhVYsFQwYqPEJQ9zan8KwZB8dG
2018-06-17 13:30:51,086 INFO     lbrynet.daemon.Downloader:186: Downloading lbry://three (348d5f) --> C:\Users\SoloDolo\Downloads\lbry downloadstest\Carvana Image Masking Challenge.pdf
2018-06-17 13:30:51,086 INFO     lbrynet.daemon.Downloader:93: Download lbry://three status changed to running
2018-06-17 13:30:57,387 INFO     lbrynet.core.client.BlobRequester:414: Offered rate 0.000000/mb accepted by 18.188.64.47
2018-06-17 13:30:57,388 INFO     lbrynet.core.client.BlobRequester:487: Requesting blob 340d8e5fc8379b5f8953ee64c603583381523456296d2cd05949500678b222c43e50fa36e530eb2790c2bebccc097f79 from 18.188.64.47:3333
2018-06-17 13:31:11,608 INFO     lbrynet.dht.hashannouncer:54: Announcing 1 blobs
2018-06-17 13:31:25,631 INFO     lbrynet.dht.hashannouncer:72: Took 14.0230000019 seconds to announce 1 of 1 attempted hashes (0.000000 hashes per second)
2018-06-17 13:31:45,319 INFO     lbrynet.core.Wallet:410: Skipping dust
2018-06-17 13:31:45,323 INFO     root:205: using buckets: [u'f7065aa34b48ccc81bbc52ee7b74fd4b10103280f2892ec894c945eca482c3c6:1']
2018-06-17 13:31:47,904 ERROR    lbryum:119: [lbryumx2.lbry.io] serving improperly sorted address histories

2018-06-17 13:34:41,608 INFO     lbrynet.dht.hashannouncer:54: Announcing 1 blobs
2018-06-17 13:34:47,081 INFO     lbrynet.dht.hashannouncer:72: Took 5.47199988365 seconds to announce 1 of 1 attempted hashes (0.000000 hashes per second)
2018-06-17 13:35:11,609 INFO     lbrynet.dht.hashannouncer:54: Announcing 1 blobs
2018-06-17 13:35:20,851 INFO     lbrynet.dht.hashannouncer:72: Took 9.24099993706 seconds to announce 1 of 1 attempted hashes (0.000000 hashes per second)
```

```
  {
    "blobs_completed": null,
    "blobs_in_stream": null,
    "channel_claim_id": null,
    "channel_name": null,
    "claim_id": "562d308abdc1f815dd2933bc681e74b410cb6c2c",
    "claim_name": "three",
    "completed": false,
    "download_directory": "C:\\Users\\SoloDolo\\Downloads\\lbry downloadstest\\",
    "download_path": "C:\\Users\\SoloDolo\\Downloads\\lbry downloadstest\\Carvana Image Masking Challenge.pdf",
    "file_name": "Carvana Image Masking Challenge.pdf",
    "key": "cf76009b20e1f12f26ef2aeffc2a54cf",
    "metadata": {
      "author": "",
      "description": "Carvana Image Masking Challenge",
      "fee": {
        "address": "bZYUVLcKXhVYsFQwYqPEJQ9zan8KwZB8dG",
        "amount": 10.0,
        "currency": "LBC",
        "version": "_0_0_1"
      },
      "language": "en",
      "license": "None",
      "licenseUrl": "",
      "nsfw": false,
      "preview": "",
      "thumbnail": "https://spee.ch/f/kaggle.jpeg",
      "title": "kaggle",
      "version": "_0_1_0"
    },
    "mime_type": "application/pdf",
    "nout": 0,
    "outpoint": "ec2265df795a7cc3c0a87180ce0e98a6a0cc5227e3f50d8cfd3940e4f6efa021:0",
    "points_paid": 0.0,
    "sd_hash": "348d5fd8faa643bfd96e5eb69196c3e2969e48f12bbac1aa4f1f6536a6dea4f1ed943f51ae758f7ee54a7371e136c659",
    "status": null,
    "stopped": true,
    "stream_hash": "bb5c6289112f039292bdb212a0b277494e29081fa4d0ff57cccf997a510d1bf5b1dc9b5ac0eeebd9930c2b0c7644f4c6",
    "stream_name": "Carvana Image Masking Challenge.pdf",
    "suggested_file_name": "Carvana Image Masking Challenge.pdf",
    "total_bytes": null,
    "txid": "ec2265df795a7cc3c0a87180ce0e98a6a0cc5227e3f50d8cfd3940e4f6efa021",
    "written_bytes": 0
  }
]
```

**Comment by tzarebczan** [comment]
Reopening, ran into this after the above commit was merged. Will keep an eye out if it happens again. File ended up in stopped status. 

```
C:\Users\SoloDolo\Desktop\latest daemon>lbrynet-cli file_list --claim_id=562d308abdc1f815dd2933bc681e74b410cb6c2c
[
  {
    "blobs_completed": null,
    "blobs_in_stream": null,
    "channel_claim_id": null,
    "channel_name": null,
    "claim_id": "562d308abdc1f815dd2933bc681e74b410cb6c2c",
    "claim_name": "three",
    "completed": false,
    "download_directory": "C:\\Users\\SoloDolo\\Downloads\\lbry downloadstest\\",
    "download_path": "C:\\Users\\SoloDolo\\Downloads\\lbry downloadstest\\Carvana Image Masking Challenge.pdf",
    "file_name": "Carvana Image Masking Challenge.pdf",
    "key": "cf76009b20e1f12f26ef2aeffc2a54cf",
    "metadata": {
      "author": "",
      "description": "Carvana Image Masking Challenge",
      "fee": {
        "address": "bZYUVLcKXhVYsFQwYqPEJQ9zan8KwZB8dG",
        "amount": 10.0,
        "currency": "LBC",
        "version": "_0_0_1"
      },
      "language": "en",
      "license": "None",
      "licenseUrl": "",
      "nsfw": false,
      "preview": "",
      "thumbnail": "https://spee.ch/f/kaggle.jpeg",
      "title": "kaggle",
      "version": "_0_1_0"
    },
    "mime_type": "application/pdf",
    "nout": 0,
    "outpoint": "ec2265df795a7cc3c0a87180ce0e98a6a0cc5227e3f50d8cfd3940e4f6efa021:0",
    "points_paid": 0.0,
    "sd_hash": "348d5fd8faa643bfd96e5eb69196c3e2969e48f12bbac1aa4f1f6536a6dea4f1ed943f51ae758f7ee54a7371e136c659",
    "status": null,
    "stopped": true,
    "stream_hash": "bb5c6289112f039292bdb212a0b277494e29081fa4d0ff57cccf997a510d1bf5b1dc9b5ac0eeebd9930c2b0c7644f4c6",
    "stream_name": "Carvana Image Masking Challenge.pdf",
    "suggested_file_name": "Carvana Image Masking Challenge.pdf",
    "total_bytes": null,
    "txid": "ec2265df795a7cc3c0a87180ce0e98a6a0cc5227e3f50d8cfd3940e4f6efa021",
    "written_bytes": 0
  }

```

**Comment by tzarebczan** [comment]
Happened to me again...

**Comment by eukreign** [comment]
@jackrobison can we add a unit/integration test for this in the new async blob exchange?
